src/common/set_up_engine.m

```diff
@@ -0,0 +1,46 @@
+function alias = set_up_engine(engine)
+% Creates a string, evaluating which imports PET engine.
+% Optionally creates also its alias (e.g. named PET, to have PET.ImageData etc.)
+
+% CCP PETMR Synergistic Image Reconstruction Framework (SIRF).
+% Copyright 2015 - 2017 Rutherford Appleton Laboratory STFC.
+% 
+% This is software developed for the Collaborative Computational
+% Project in Positron Emission Tomography and Magnetic Resonance imaging
+% (http://www.ccppetmr.ac.uk/).
+% 
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+% http://www.apache.org/licenses/LICENSE-2.0
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+
+    narginchk(1,1);
+    try
+        libload_sirf();
+        eval(['libload_' lower(engine)])
+    catch me
+        fprintf(me.message)
+        error('package %s failed to load\n', engine)
+    end
+    
+    if nargout == 0 
+        return;
+    end
+    
+    %% For returning as an alias
+    
+    % Get the folder containing the engine
+    path = [fileparts(mfilename('fullpath')) '/+m' engine '/'];
```

Comment:
likely not portable to Windows. Use 'fullfile' or 'filesep'

Reply:
ok, my new implementation moves the need of the `/` to here:
```
files = dir([path '/*.m']);
```
From the [Matlab doc](https://uk.mathworks.com/help/matlab/ref/dir.html), I think this is ok:
> For example, dir \*/*.txt lists all files with a txt extension exactly one folder under the current folder

Reply:
sorry. why not
```
files = dir(fullfile(path, '*.m'));
```

Reply:
much better!